Userland/Libraries/LibMarkdown/Text.cpp

```diff
@@ -169,24 +169,49 @@ Text Text::parse(StringView const& str)
     return text;
 }
 
+static bool flanking(StringView const& str, size_t start, size_t end, int dir)
+{
+    ssize_t next = ((dir > 0) ? end : start) + dir;
+    if (next < 0 || next >= (ssize_t)str.length())
+        return false;
+
+    if (isspace(str[next]))
+        return false;
+
+    if (!ispunct(str[next]))
+        return true;
+
+    ssize_t prev = ((dir > 0) ? start : end) - dir;
+    if (prev < 0 || prev >= (ssize_t)str.length())
+        return true;
+
+    return isspace(str[prev]) || ispunct(str[prev]);
+}
+
 Vector<Text::Token> Text::tokenize(StringView const& str)
 {
     Vector<Token> tokens;
     StringBuilder current_token;
 
-    auto flush_token = [&](bool left_flanking, bool right_flanking, bool is_run) {
+    auto flush_run = [&](bool left_flanking, bool right_flanking, bool punct_before, bool punct_after, bool is_run) {
```

Comment:
This commit seems to fix-up a previous commit. Can this be squashed into the first LibMarkdown commit, or is that a bad idea for some reason? (This is an actual question; maybe I just overlooked something.)

Reply:
It does change the name, but it also adds a distinct feature (specifically, the punctuation rules for flanking `_`s). I don't think that it would be beneficial.